wagtail/models/i18n.py

```diff
@@ -174,13 +174,182 @@ def is_active(self) -> bool:
+        ``prefetch_related`` are not retained. If you want to use them on the resulting
+        queryset, you should apply them after ``.localized()``. However, you can use
+        `annotate` on the original queryset and use its value for filtering. Just be
+        aware that the annotate column will not be available on the returned queryset.
+        """
+        # Skip localization if i18n is not enabled. This behavior is consistent with
+        # the behavior of the `localized` property on `TranslatableMixin`.
+        if not getattr(settings, "WAGTAIL_I18N_ENABLED", False):
+            return self
+
+        # Get all instances that are available in the active locale. We can find these
+        # by getting all model instances that have a translation key from the original
+        # queryset and that are available in the active locale.
+        active_locale = Locale.get_active()
+        original_translation_keys = self.values_list("translation_key", flat=True)
+        translated_instances = self.model._default_manager.filter(
+            locale_id=pk(active_locale),
+            translation_key__in=original_translation_keys,
+        )
+
+        # Don't consider draft translations. If a translation is in draft, we want to
+        # use the original instance instead. To do so, we exclude draft translations.
+        # This only applies if the model has a `live` field. We allow bypassing this
+        # behavior by passing `include_draft_translations=True`.
+        from wagtail.models import DraftStateMixin
+
+        if issubclass(self.model, DraftStateMixin) and not include_draft_translations:
+            translated_instances = translated_instances.exclude(live=False)
+
+        if include_only_translations:
+            # If we only want to include translations, we can use the translated
+            # instances as the localized queryset.
+            localized_queryset = translated_instances
+        else:
+            # Otherwise, we need to combine the translated instances with the
+            # untranslated instance.
+
+            # Get all instances that are not available in the active locale, these are
+            # the untranslated instances. We can find these by excluding the translation
+            # keys for which translations exist from the original queryset.
+            translated_translation_keys = translated_instances.values_list(
+                "translation_key", flat=True
+            )
+            untranslated_instances = self.exclude(
+                translation_key__in=translated_translation_keys,
+            )
+
+            # Combine the translated and untranslated querysets to get the localized
+            # queryset.
+            localized_queryset = self.model._default_manager.filter(
+                models.Q(pk__in=translated_instances)
+                | models.Q(pk__in=untranslated_instances)
+            )
+
```

Comment:
Hey @tbrlpld, this works to get the annotations back into the localized queryset:

```suggestion
if annotations := queryset.query.annotations:
    localized_queryset = localized_queryset.annotate(**annotations)

```

However, the values will be based on the localized queryset, so that the annotations will be updated on the translated instances. Maybe it's worth highlighting that.

The good thing is that it definitely allows us to retain the original order of the queryset when the ordering was done through an annotation.

Reply:
> However, the values will be based on the localized queryset, so that the annotations will be updated on the translated instances. Maybe it's worth highlighting that.

That should be expected, no? Annotations are make on a per-row basis, so a different instance should be expected to have a different value.

Reply:
I guess your case would still require preserving the original ordering.

Reply:
Thanks @jhonatan-lopes. I have incorporated that. I had to make some extra additions to also make it work for [aliases](https://docs.djangoproject.com/en/4.2/ref/models/querysets#django.db.models.query.QuerySet.alias).